ecom_test/tests/backend_tests/product_reviews/test_products_reviews_statuses.py

```diff
@@ -0,0 +1,128 @@
+import pytest
+from ecom_test.src.generic_helpers.generic_product_helper import GenericProductHelper
+from ecom_test.src.generic_helpers.generic_product_review_helper import GenericProductReviewsHelper
+from ecom_test.src.utilities.generic_utilities import generate_random_string, generate_random_email_and_password
+import logging as logger
+
+
+@pytest.fixture(scope="function")
+def setup_teardown(request):
+    # Create instances of the necessary helper classes
+    request.cls.product_helper = GenericProductHelper()
+    request.cls.product_review_helper = GenericProductReviewsHelper()
+
+    # Create a product and store its ID for reference
+    request.cls.product = request.cls.product_helper.create_a_product()
+    request.cls.rating_count_before = request.cls.product['rating_count']
+    request.cls.product_id = request.cls.product['id']
+
+    yield request.cls.product_helper, request.cls.product_review_helper, request.cls.product_id
+    # They become available to the test function that consumes the fixture
+
+    # Perform cleanup by deleting the product created for the test
+    logger.info(f"Running teardown. Deleting product id: {request.cls.product_id}")
+    request.cls.product_helper.delete_a_product(request.cls.product_id)
+
+
+@pytest.mark.usefixtures("setup_teardown")
+@pytest.mark.smoke
+class TestProductsReviewsSmoke(object):
+    """
+    Test suite for smoke testing product reviews functionality.
+    """
+
+    @pytest.mark.status
+    @pytest.mark.parametrize("review_status", [
+        pytest.param('hold', marks=[pytest.mark.tcid54]),  # Test case ID: 54, review status: hold
+        pytest.param('spam', marks=[pytest.mark.tcid55]),  # Test case ID: 55, review status: spam
+        pytest.param('unspam', marks=[pytest.mark.tcid56]),  # Test case ID: 56, review status: un-spam
+        pytest.param('trash', marks=[pytest.mark.tcid57]),  # Test case ID: 57, review status: trash
+        pytest.param('untrash', marks=[pytest.mark.tcid58])  # Test case ID: 58, review status: un-trash
+    ])
+    def test_verify_create_review_with_different_statuses(self, setup_teardown, review_status, rating=5):
```

Comment:
do you have pass 'setup_teardown' even if you are using @pytest.mark.usefixtures? I am not sure. can you try without passing in setup_teardown fixture as a parameter?

Reply:
Yes, I tried initially but it did not work well and also did not give auto suggestions for the function (when I call). By adding setup_teardown I can track any function. 
![image](https://github.com/Rohanksingh18/ecom_site_automation/assets/126942008/921a918d-928e-46b1-9551-f0560cdd05d8) 
without setup_teardown: 
![image](https://github.com/Rohanksingh18/ecom_site_automation/assets/126942008/109ae88c-c624-4884-8b86-1d3fd4d676ae)